ZeekPluginStatic.cmake

```diff
@@ -73,9 +73,18 @@ function (zeek_add_static_plugin ns name)
         target_include_directories(${target_name} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/src)
     endif ()
 
-    # Add user-defined extra include directories.
+    # Add user-defined extra include directories. If a path is outside of the current
+    # project source dir, add it as a system path so that clang-tidy can easily ignore
+    # it.
     if (FN_ARGS_INCLUDE_DIRS)
-        target_include_directories(${target_name} PRIVATE ${FN_ARGS_INCLUDE_DIRS})
+        foreach (_include_dir ${FN_ARGS_INCLUDE_DIRS})
+            cmake_path(IS_PREFIX PROJECT_SOURCE_DIR "${_include_dir}" _is_project_prefixed)
```

Comment:
Is this for not doing it explicitly like done here?

https://github.com/zeek/zeek/blob/53c56491eec79fe8dffd38fef2429ac7ac505d1d/src/cluster/websocket/auxil/CMakeLists.txt#L9-L15

What are include dirs causing issues?

Reply:
The difference is that adding the directories as system dirs allows me to ignore them all at once, instead of individually having to SKIP_LINTING on each file we might include from somewhere. For example, I can add `/opt/homebrew/include` on macOS and catch a lot of external files that we'd include.

As for what dirs were causing issues, I ran into openssl, krb5, hiredis, and a few others. Yes, I could extend the regex in .clang-tidy to include these (and I did at first), but this method makes more sense to me.